### PR TITLE
Moved code from ckeditor5 main repository

### LINKED ISSUE #1: Move code from ckeditor5 repository
As described here https://github.com/ckeditor/ckeditor5/issues/218.

### PR DESCRIPTION
**Pull request by szymonkups**
Fixes #1.

### CODE REVIEW

File: lib/docs/index.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+			const esnextBuildPath = path.join( config.ROOT_DIR, config.BUILD_DIR, 'esnext' );
+			const jsDocConfig = {
+				opts: {
+					encoding: 'utf8',
+					destination: path.join( config.BUILD_DIR, 'docs' ),
+					recurse: true,
+					access: 'all'
+				},
+				plugins: [
+					path.resolve( __dirname, '../../node_modules/jsdoc/plugins/markdown' ),
```
Comment (maxbarnas) [inline lib/docs/index.js]: Are you sure this is the right path? If this module is inside `node_modules/` dir, then there's no need to specify that directory. Just couple of `../` and the name of package `jsdoc` should do the trick.

Comment (szymonkups) [inline lib/docs/index.js]: Thanks for pointing this out. It looks like for `jsdoc` native plugins it is enough to pass `plugins/markdown`.

Comment (maxbarnas) [inline lib/docs/index.js]: You mean that we don't need this?

Comment (szymonkups) [inline lib/docs/index.js]: I mean that we don't need to specify whole path to plugins from `jsdoc` directory. It only requires `plugins/[pluginname]`.

File: lib/builder/index.js [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+const gutil = require( 'gulp-util' );
+const filter = require( 'gulp-filter' );
+const { build, tools } = require( 'ckeditor5-dev-utils' );
+
+/**
+ * Exports function returning builder tasks.
+ *
+ * @param {Object} config
+ * @param {String} config.ROOT_DIR Root of main project repository.
+ * @param {String} config.BUILD_DIR Relative path to build directory.
```
Comment (maxbarnas) [inline lib/builder/index.js]: The `path.join()` below suggests that `BUILD_DIR` is not path, but directory name. Especially that it is defined like directory name in `ckeditor5/gulpfile.js`.

**Pull request merged**